### PR TITLE
fix(oas): improve `consumes/produces` configuration handling

### LINKED ISSUE #210: OAS: Missing media-type configuration leads to swagger converter error
### Description

Converting of specification to HAR may results in error when both operation and root `consumes` nodes missing. Nearly the same case for `produces` configuration, converter may not take into account the root `produces` configuration of the operation level configuration which overrides the root one.

Here it is required to improve converter when it handles `consumes/produces` configuration.

### Steps to reproduce

Pass the `my-schema.yaml` (see below) to the `oas2har` as follows:

```ts
import { oas2har } from '@har-sdk/oas';
import { readFile } from 'fs';
import { promisify } from 'util';
import { load } from 'js-yaml';

const content = await promisify(readFile)(
  './my-schema.yaml',
  'utf8'
);

const result = oas2har(load(content) as OpenAPIV2.Document);
```

<details>
<summary>my-schema.yaml</summary>

```
swagger: '2.0'
info:
  title: Future is Bright
  version: 1.0.0
host: brokencrystals.com
basePath: /api/v1
schemes:
  - https
paths:
  '/dummy1':
    post:
      parameters:
        - in: body
          name: body
          schema:
            $ref: '#/definitions/requestBody'
      responses:
        200:
          description: successful operation
          schema: {}
definitions:
  requestBody:
    type: object
    properties:
      id:
        type: integer
```
</details>

### Actual result

Error produced:

```
contentType.split is not a function
TypeError: contentType.split is not a function
    at Oas2BodyConverter.encodeValue 
    at Oas2BodyConverter.encodePayload 
    at Oas2BodyConverter.convertBody 
    at Oas2BodyConverter.convert 
    at DefaultConverter.convertPart 
    at DefaultConverter.createHarEntry
...
```

### Expected result

Conversion performed successfully, `application-json` media type used for `content-type` header and `*/*` media type used for `accept` header.

### PR DESCRIPTION
**Pull request by ostridm**
closes #210

### CODE REVIEW

**Review by derevnjuk — CHANGES_REQUESTED** [review]

File: packages/oas/tests/fixtures/params-form-data.swagger.result.json [hunk 1]
```diff
@@ -6,15 +6,19 @@
       {
         "name": "content-type",
         "value": "multipart/form-data"
+      },
+      {
+        "name": "accept",
+        "value": "application/json"
```
[the remark this answers is not in the document]
Comment (ostridm) [reply packages/oas/tests/fixtures/params-form-data.swagger.result.json:12]: This particular case has explicit root level `produces`, see below. Introduced default `*/*` value for missing `produces` configuration.

``` yaml
consumes:
  - application/json
  - multipart/form-data
produces:
  - application/json
paths:
  /example:
    post:
      operationId: addExample
      consumes:
        - multipart/form-data
      parameters:
        - name: file
          in: formData
          type: file
          required: true
```

File: packages/oas/tests/fixtures/github.swagger.result.json
Comment (pmstss) [inline packages/oas/tests/fixtures/github.swagger.result.json:1]: it's hard to check relevant changes especially in this 200KB file due to reordered properties; please rework fixtures result formatting to keep previous ordering

**Review by pmstss — CHANGES_REQUESTED** [review]

File: packages/oas/tests/fixtures/github.swagger.result.json
Comment (ostridm) [reply packages/oas/tests/fixtures/github.swagger.result.json:1]: reworked, please proceed

File: packages/oas/src/converter/parts/headers/HeadersConverter.ts [hunk 2]
```diff
@@ -32,7 +39,15 @@ export abstract class HeadersConverter<T extends OpenAPI.Document>
     const headers: Header[] = [];
     const pathObj = getOperation(this.spec, path, method);
 
-    headers.push(...this.createContentTypeHeaders(pathObj));
+    if (
+      this.CONTENT_TYPE_METHODS.includes(method.toLowerCase()) &&
+      !headers.find(
+        (header) => header.name.toLowerCase() === this.CONTENT_TYPE_HEADER
+      )
```
Comment (pmstss) [inline packages/oas/src/converter/parts/headers/HeadersConverter.ts:46]:
```suggestion
      !headers.some(
        (header) => header.name.toLowerCase() === this.CONTENT_TYPE_HEADER
      )
```

**Review by pmstss — CHANGES_REQUESTED** [review]
thanks, looks good, please consider following improvement in addition to commented in code:
- please add explicit root level `consumes` fixture
- please add fixtures for `produces` cases (root-level, operation-level, overriding)

**Review by pmstss — APPROVED** [review]

**Review by derevnjuk — APPROVED** [review]

**Pull request merged**